sensu/manifests/init.pp

```diff
@@ -0,0 +1,335 @@
+#     as a file reference, as you'd normally configure sensu.
+#   Default: undef
+#
+# [*rabbitmq_reconnect_on_error*]
+#   Boolean. In the event the connection or channel is closed by RabbitMQ, attempt to automatically
+#     reconnect when possible. Default set to fault its not guaranteed to successfully reconnect.
+#   Default: false
+#   Valid values: true, false
+#
+# [*redis_host*]
+#   String.  Hostname of redis to be used by sensu
+#   Default: localhost
+#
+# [*redis_port*]
+#   Integer.  Redis port to be used by sensu
+#   Default: 6379
+#
+# [*redis_password*]
+#   String.  Password to be used to connect to Redis
+#   Default: undef
+#
+# [*reddis_reconnect_on_error*]
+#   Boolean. In the event the connection or channel is closed by Reddis, attempt to automatically
+#     reconnect when possible. Default set to fault its not guaranteed to successfully reconnect.
+#   Default: false
+#   Valid values: true, false
+#
+# [*api_bind*]
+#   String.  IP to bind api service
+#   Default: 0.0.0.0
+#
+# [*api_host*]
+#   String.  Hostname of the sensu api service
+#   Default: localhost
+#
+# [*api_port*]
+#   Integer. Port of the sensu api service
+#   Default: 4567
+#
+# [*api_user*]
+#   String.  Password of the sensu api service
+#   Default: undef
+#
+# [*api_password*]
+#   String. Password of the sensu api service
+#   Default: undef
+#
+# [*subscriptions*]
+#   Array of strings.  Default suscriptions used by the client
+#   Default: []
+#
+# [*client_address*]
+#   String.  Address of the client to report with checks
+#   Default: $::ipaddress
+#
+# [*client_name*]
+#   String.  Name of the client to report with checks
+#   Default: $::fqdn
+#
+# [*client_custom*]
+#   Hash.  Custom client variables
+#   Default: {}
+#
+# [*client_keepalive*]
+#   Hash.  Client keepalive config
+#   Default: {}
+#
+# [*safe_mode*]
+#   Boolean.  Force safe mode for checks
+#   Default: false
+#   Valid values: true, false
+#
+# [*plugins*]
+#   String, Array of Strings.  Plugins to install on the node
+#   Default: []
+#
+# [*plugins_dir*]
+#   String. Puppet url to plugins directory
+#   Default: undef
+#
+# [*purge_config*]
+#   Boolean.  If unused configs should be removed from the system
+#   Default: false
+#   Valid values: true, false
+#
+# [*use_embedded_ruby*]
+#   Boolean.  If the embedded ruby should be used
+#   Default: false
+#   Valid values: true, false
+#
+# [*rubyopt*]
+#   String.  Ruby opts to be passed to the sensu services
+#   Default: ''
+#
+# [*gem_path*]
+#   String.  Paths to add to GEM_PATH if we need to look for different dirs.
+#   Default: ''
+#
+# [*log_level*]
+#   String.  Sensu log level to be used
+#   Default: 'info'
+#   Valid values: debug, info, warn, error, fatal
+#
+# [*init_stop_max_wait*]
+#   Integer.  Number of seconds to wait for the init stop script to run
+#   Default: 10
+#
+class sensu (
+  $version                     = 'latest',
```

Comment:
Are we using the parameters here? If yes, we need to move them to hiera as they contain username and password parameters.